docs/src/upgrade/release-3.7.x.asciidoc

```diff
@@ -29,7 +29,68 @@ complete list of all the modifications that are part of this release.
 
 === Upgrading for Users
 
+==== String Manipulation Steps
+Additional String manipulation steps have been added to this version: `asString()`, `length()`, `toLower()`, `toUpper()`.
 
+The following example demonstrates the use of a closure to perform the above functions:
+[source,text]
+----
+gremlin> g.V().hasLabel("person").values("age").map{it.get().toString()}
+==>29
+==>27
+==>32
+==>35
+gremlin> g.V().values("name").map{it.get().length()}
+==>5
+==>5
+==>3
+==>4
+==>6
+==>5
+gremlin> g.inject("TO", "LoWeR", "cAsE").map{it.get().toLowerCase()}
+==>to
+==>lower
+==>case
+gremlin> g.V().values("name").map{it.get().toUpperCase()}
+==>MARKO
+==>VADAS
+==>LOP
+==>JOSH
+==>RIPPLE
+==>PETER
+----
+With these additional steps this operation can be performed with standard Gremlin syntax:
+[source,text]
+----
+gremlin> g.V().hasLabel("person").values("age").asString()
+==>29
+==>27
+==>32
+==>35
+gremlin> g.V().values("name").length()
+==>5
+==>5
+==>3
+==>4
+==>6
+==>5
+gremlin> g.inject("TO", "LoWeR", "cAsE").toLower()
+==>to
+==>lower
+==>case
+gremlin> g.V().values("name").toUpper()
+==>MARKO
+==>VADAS
+==>LOP
+==>JOSH
+==>RIPPLE
+==>PETER
+----
+See: link:https://tinkerpop.apache.org/docs/x.y.z/reference/#asString-step[reference]
```

Comment:
(nitpick) Maybe use different text for these links than just `reference` for every step? Like this, it only shows 4 links with the text "reference". How about "asString()-step" and so on?
That's how it was done for `mergeV()` and `mergeE()`: https://github.com/apache/tinkerpop/blob/8eb1cf99d3393c35ad73c5516e6ac15697dcf973/docs/src/upgrade/release-3.6.x.asciidoc#mergev-and-mergee